json/src/value.rs

```diff
@@ -625,6 +625,137 @@ impl<I> ops::Index<I> for Value where I: Index {
     }
 }
 
+impl PartialEq<str> for Value {
+    fn eq(&self, other: &str) -> bool {
+        self.as_str().map_or(false, |s| s == other)
+    }
+}
+
+impl<'a> PartialEq<&'a str> for Value {
+    fn eq(&self, other: &&str) -> bool {
+        self.as_str().map_or(false, |s| s == *other)
+    }
+}
+
+impl PartialEq<Value> for str {
+    fn eq(&self, other: &Value) -> bool {
+        other.as_str().map_or(false, |s| s == self)
+    }
+}
+
+impl<'a> PartialEq<Value> for &'a str {
+    fn eq(&self, other: &Value) -> bool {
+        other.as_str().map_or(false, |s| s == *self)
+    }
+}
+
+impl PartialEq<String> for Value {
+    fn eq(&self, other: &String) -> bool {
+        self.as_str().map_or(false, |s| s == other)
+    }
+}
+
+
+impl PartialEq<Value> for String {
+    fn eq(&self, other: &Value) -> bool {
+        other.as_str().map_or(false, |s| s == self)
+    }
+}
+
+macro_rules! partialeq_ixx {
```

Comment:
you can merge all these macros, by passing the `as_*` function name and the `as *` cast type as macro arguments.